sagemaker-neo-notebooks/edge/cpp-integration/tutorial.cc

```diff
@@ -0,0 +1,622 @@
+  if (isSuccess == false) {
+    throw std::runtime_error("Compilation timeout");
+  }
+  cout << "Done!" << endl;
+}
+
+void GetCompiledModelFromNeo(string &bucket_name, string &model_name,
+                             string &target, string &compiled_filename) {
+  const string output_path = "output/" + model_name + "-" + target;
+
+  Aws::String aws_bucket_name(bucket_name.c_str(), bucket_name.size());
+  Aws::String aws_object_name(output_path.c_str(), output_path.size());
+
+  // download s3 object
+  Aws::S3::S3Client s3_client = getS3Client();
+
+  Aws::S3::Model::GetObjectRequest object_request;
+  object_request.SetBucket(aws_bucket_name);
+  object_request.SetKey(aws_object_name);
+
+  auto get_object_outcome = s3_client.GetObject(object_request);
+  if (!get_object_outcome.IsSuccess()) {
+    auto error = get_object_outcome.GetError();
+    const std::string error_str = getErrorMessage(error);
+    throw std::runtime_error("GetModel error: " + error_str);
+  } else {
+    auto &model_file = get_object_outcome.GetResultWithOwnership().GetBody();
+
+    // download the sample file
+    const char *output_filename = compiled_filename.c_str();
+    std::ofstream output_file(output_filename, std::ios::binary);
+    output_file << model_file.rdbuf();
+  }
+}
+
+void DownloadNpyData() {
+  Aws::S3::S3Client s3_client = getS3Client();
+
+  // this is a preprocess image file in our sample directory
+  // this can be changed to any s3 bucket you specified
+  Aws::String aws_bucket_name = "neo-ai-dlr-test-artifacts";
+  Aws::String aws_object_name = "test-data/dog.npy";
+  std::string filename = "dog.npy";
+
+  Aws::S3::Model::GetObjectRequest object_request;
+  object_request.SetBucket(aws_bucket_name);
+  object_request.SetKey(aws_object_name);
+
+  auto get_object_outcome = s3_client.GetObject(object_request);
+  if (!get_object_outcome.IsSuccess()) {
+    auto error = get_object_outcome.GetError();
+    const std::string error_str = getErrorMessage(error);
+    throw std::runtime_error("DownloadNpyData error: " + error_str);
+  } else {
+    auto &image_file = get_object_outcome.GetResultWithOwnership().GetBody();
+    const char *output_filename = filename.c_str();
+    std::ofstream output_file(output_filename, std::ios::binary);
+    output_file << image_file.rdbuf();
+  }
+}
+
+template <typename T>
+int GetPreprocessNpyFile(string npy_filename,
+                         std::vector<unsigned long> &input_shape,
+                         std::vector<T> &input_data) {
+  bool fortran_order;
+  npy::LoadArrayFromNumpy(npy_filename, input_shape, fortran_order, input_data);
+  return 0;
+}
+
+void RunInference(const std::string &compiled_model,
+                  const std::string &npy_name) {
+  DLRModelHandle handle;
+
+  std::cout << "CreateDLRModel" << std::endl;
+  int dev_type = 1;  // cpu == 1
+  int dev_id = 0;
+  char *model_path = const_cast<char *>(compiled_model.c_str());
+  CreateDLRModel(&handle, model_path, dev_type, dev_id);
```

Comment:
We do not need to cast `c_str` output to `char*`. `CreateDLRModel` param is `const char* model_path`, try the following:
```
CreateDLRModel(&handle, compiled_model.c_str(), dev_type, dev_id);
```